website/static/js/filepage/index.js

```diff
@@ -194,6 +195,49 @@ var FileViewPage = {
                 ], '.')
             ]);
         }
+        var sharePopover =  {view: function() { return m('button.btn.btn-sm.btn-primary.file-share', {onclick: function () {
```

Comment:
In general, design components for re-usability; declaring them inside view functions makes it impossible to reuse them.